tensorflow/core/kernels/data/dataset_test_base.h

```diff
@@ -162,6 +164,27 @@ struct IteratorSaveAndRestoreTestCase {
   std::vector<Tensor> expected_outputs;
 };
 
+// Creates a tensor with the specified dtype, shape, and value.
+template <typename T>
+static Tensor CreateTensor(const TensorShape& input_shape,
+                           const gtl::ArraySlice<T>& input_data) {
+  Tensor tensor(DataTypeToEnum<T>::value, input_shape);
+  test::FillValues<T>(&tensor, input_data);
+  return tensor;
+}
+
+// Creates a vector of tensors with the specified dtype, shape, and values.
+template <typename T>
+std::vector<Tensor> CreateTensors(
+    const TensorShape& shape, const std::vector<gtl::ArraySlice<T>>& values) {
```

Comment:
`gtl::ArraySlice<T>` can not be `const`, which will cause the error `invalid conversion from 'const void*' to 'void*'` in some compiler.